README.md

```diff
@@ -185,21 +185,66 @@ sponsoring the project.
 
 ### Type signatures
 
-[Hindley-Milner][Guide:HM] type signatures are used to document functions, with
-one addition: In order to document *methods* as well as static functions, we
-prefix every signature with `.` or `#`. This indicates whether the function
-lives on the object (`.function`), or its prototype (`#method`). We use `~>` to
-separate the implicit `this` argument from the other, explicit, arguments.
+[Hindley-Milner][Guide:HM] type signatures are used to document functions. You
+might encounter some additional syntax that we use to describe JavaScript
+specific stuff, like [methods](#squiggly-arrows) or functions that take
+[multiple arguments at once](#brackets).
+
+#### Squiggly Arrows
+
+In order to document *methods* as well as static functions, use the squiggly
+arrow (`~>`). This separates the implicit `this` argument from the other,
+explicit, arguments. For example, the following line signifies a *function*,
+because it doesn't have a squiggly arrow in its signature:
+
+```hs
+map :: (b -> c) -> Future a b -> Future a c
+```
+
+Whereas the next example is a *method*. It needs a `this` as indicated by the
+use of the squiggly arrow. The type on the left of the squiggly arrow also gives
+away where the method lives. In this case it's on the prototype of `Future`:
+
+```hs
+map :: Future a b ~> (b -> c) -> Future a c
+```
+
+#### Brackets
+
+Most functions exposed by Fluture are curried. This is reflected in their type
+signatures by using an arrow at each step where partial application is possible.
+For example, the following line signifies a *curried* function, because it has
+an arrow in between every function argument:
+
+```hs
+add :: Number -> Number -> Number
+```
+
+We could have chosen to write the above line with "groups of one argument", but
+we usually leave the grouping brackets out for brevity:
+
+```hs
+add :: (Number) -> (Number) -> Number
+```
+
+In order to document functions and methods that are *not* curried, we use
+grouping to show which arguments have to be provided at the same time:
+
+```hs
+add :: (Number, Number) -> Number
+```
 
 #### Types
 
 You'll find that some signatures refer to concrete types, such as `Future`.
 This is reference of the types used throughout the documentation:
 
+- **()** - Nothing, also known as "unit". Corresponds to `undefined`.
 - **Future** - Instances of Future provided by [compatible versions](#casting-futures) of Fluture.
 - **ConcurrentFuture** - [Concurrified][concurrify] Futures ([`Future.Par`](#concurrentfuture)).
 - **Promise** - Values which conform to the [Promises/A+ specification][7].
 - **Nodeback a b** - A Node-style callback; A function of signature `(a | Nil, b) -> ()`.
+- **Tuple a b** - An array with exactly two elements: `[a, b]`.
```

Comment:
Consider using `Pair a b` instead to match Sanctuary (see [`S.pairs`][1], for example).


[1]: https://sanctuary.js.org/#pairs